Reactjs

bsshashank opened the pull request:
Hi Andreas,

I've implemented the SettingsManager and updated the PluginsManager.

Commit:
Implemented Settings Manager to just display settings for all the installed plugins

Commit:
Added delete button to the plugin card and performed code refactoring for object presentation

Commit:
Added functionality to Plugin Manager for deleting directory and integrated it with Settings Manager to uninstall a plugin

Commit:
Changed Settings manager into a folder to modify it to a reactor app and added path.join to the SettingsManager

Commit:
Implemented SettingsManager to make use of reactjs and implemented new feature to unistall a plugin from the settings page

Commit:
Added comments to the PluginManager, SettingsManager and a few other files.

Commit:
Merge branch 'reactjs' of https://github.com/appelgriebsch/electron-boilerplate into reactjs

Commit:
Added drag and drop functionality to install new plugins

Commit:
Added drag and drop functionality to install new plugins

Commit:
Revert "Added drag and drop functionality to install new plugins"

This reverts commit a78941114c92688625b10d9aaf9dd9a125241509.

Commit:
Fixed bugs with the install plugin feature

appelgriebsch commented on app/components/PluginCard.js:
```diff
@@ -0,0 +1,4 @@
+'use strict';Object.defineProperty(exports,'__esModule',{value:true});var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if('value'in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor)}}return function(Con ...
```
Pls. do not check-in generated code... The .jsx files are sufficient!

bsshashank replied:
Okay, I'll add it into the .gitignore file

appelgriebsch commented on app/main.html:
```diff
@@ -21,6 +21,6 @@
 
     <script type="application/javascript" src="node_modules/react-mdl/extra/material.min.js"></script>
-    <script type="application/javascript" src="main.js"></script>
+    <script type="application/javascript" src="app.js"></script>
```
Is main.js no longer needed? Why not changing main.js instead of introducing app.js(x)?

bsshashank replied:
Sorry, I forgot to integrate it with main.js. I didn't want to change main.js directly. I'll make the changes.

appelgriebsch reviewed (commented):
Please also remove the .asar files from plugins/ directory. We should discuss providing some ``example´´ projects for them...

Commit:
"Modified the app structure and moved the store initializations to app.jsx
from RoutesManager. Deleted .asar packages. Fixed plugin installation
features.

bsshashank committed:
Merged with upstream master
